### PR TITLE
feat: Add `web` auth type

### PR DESCRIPTION
**Pull request by jumoel**
## What
Add the more generic `web` for the registry to use, instead of the implementation-specific `webauthn` authentication type.

## Why
It's more generic and sounds better.

## References
- Closes https://github.com/github/npm/issues/5499

### CODE REVIEW

**Comment by ljharb** [comment]
wouldn’t this be a breaking change? Also there’s tons of kinds of web auth that isn’t webauthn.

**Comment by jumoel** [comment]
@ljharb:

> wouldn’t this be a breaking change?

Technically yes, but it’s never been enabled on the registry, except in staging, so it won’t be a problem.

We’ve already enabled the new header on the registry (and kept support for the old one).

> Also there’s tons of kinds of web auth that isn’t webauthn.

Which is one of the reasons it’s changing. It’s not only webauthn that’s supported.

**Comment by ljharb** [comment]
What about private registries that may have implemented support in the meantime?

**Comment by darcyclarke** [comment]
@jumoel @MylesBorins as @ljharb notes, this is a breaking change now so we should probably just keep this as `"webauthn"` & not introduce another value for `auth-type` in the same major (ref. https://docs.npmjs.com/cli/v8/using-npm/config#auth-type).

**Review by darcyclarke — CHANGES_REQUESTED** [review]
!Breaking change

**Comment by jumoel** [comment]
@darcyclarke what if we just add web as another alias?

**Comment by MylesBorins** [comment]
> What about private registries that may have implemented support in the meantime?

There is 0 chance anyone has implemented this. If they had they likely would have reached out to clarify why the field is called webauthn.

The auth flow we are implementing is to "authenticate via website" not specific to WebAuthn. Our enhanced 2FA experience was launched explicitly [as a beta](https://github.blog/changelog/2022-05-10-enhanced-2fa-experience-for-npm-accounts-public-beta/). IMHO such beta services do not have to adhere to Semver (similar to the experimental status in Node.js). With that said, I recognize there is ambiguity here and we should be thoughtful about SemVer, but I also want to be pragmatic about committing the CLI to permanent legacy behavior for something that isn't currently supported by any registry.

I think we need to introduce `--auth-type web` in npm 8 for this feature with the proper header being sent to the registry, it does not make sense for us to have incorrect functionality or internals due to miscommunication during product design.

It seems like what we do need to decide is "what to do with the `--auth-type webauthn` flag. There seem to be two options

1. Remove `--auth-type webauthn` in npm 8

This is the cleanest approach imho, and operates under the assumption that the new 2FA experience and related functionality are beta. It is being a bit cavalier about SemVer and has a non-zero risk of breaking things... but I would bet a steak dinner that nothing breaks and this is a much cleaner solution

2. Deprecate `--auth-type webauthn` in npm 8 and remove it in npm 9.

In this approach we doc deprecate + warn for webauthn and remove it in 9.

---

Thoughts?

File: docs/content/using-npm/config.md [hunk 1]
```diff
@@ -1908,13 +1908,13 @@ When set to `dev` or `development`, this is an alias for `--include=dev`.
...
-* Type: "legacy", "webauthn", "sso", "saml", or "oauth"
-* DEPRECATED: The SSO/SAML/OAuth methods are deprecated and will be removed in
-  a future version of npm in favor of web-based login.
+* Type: "legacy", "web", "sso", "saml", "oauth", or "webauthn"
+* DEPRECATED: The SSO/SAML/OAuth/Webauthn methods are deprecated and will be
+  removed in a future version of npm in favor of web-based login.
 
 What authentication strategy to use with `adduser`/`login`.
 
-Pass `webauthn` to use a web-based login.
```
Comment (MylesBorins) [inline docs/content/using-npm/config.md]: Same as above, I think we can just drop all documentation related to `webauthn`

**Comment by darcyclarke** [comment]
@jumoel are you going to be updating this PR from the feedback above? Based on internal discussions & this thread, I believe the path forward here is to leave `webauthn` as a valid `auth-type` & just add `web` separately (the two should not coerce to eachother & you'll have to update this PR to **not** remove `webauthn`). 

Our team can/will deprecate & remove the `webauthn` value in future versions.

**Comment by MylesBorins** [comment]
> I believe the path forward here is to leave webauthn as a valid auth-type & just add web separately (the two should not coerce to eachother & you'll have to update this PR to not remove webauthn).

@darcyclarke I believe that is what this PR does in it's current implementation. We can ignore my comments about updating docs

**Comment by wraithgar** [comment]
This is going to collide with https://github.com/npm/cli/pull/5149 when that one lands.  We'll have to rebase and update the copy, since we have switched to being able to deprecate certain config values instead of the entire config option.

File: lib/utils/config/definitions.js [hunk 2]
```diff
@@ -239,15 +239,14 @@ define('audit-level', {
...
   // is deprecated
   description: `
-    NOTE: auth-type values "sso", "saml", and "oauth" will be removed in a future version.
-    
-    What authentication strategy to use with \`login\`.
+    NOTE: auth-type values "sso", "saml", "oauth", and "webauthn" will be
+    removed in a future version.
 
-    Pass \`webauthn\` to use a web-based login.
+    What authentication strategy to use with \`login\`.
```
Comment (MylesBorins) [inline lib/utils/config/definitions.js:249]:
```suggestion
     Pass \`web\` to use a web-based login.
```

**Review by MylesBorins — APPROVED** [review]
LGTM with a tiny nit

File: lib/utils/config/definitions.js (hunk 2 above)
Comment (wraithgar) [reply lib/utils/config/definitions.js:249]: I don't think removing the actual description is what we want here.

**Pull request merged**